Feature EPL-1454: How to create a pick and execute process (#189)
etendosoftware/docs: docs/developer-guide/etendo-classic/how-to-guides/how-to-create-a-pick-and-execute-process.md

---
tags: 
  - PickandExecute
  - Process definition
  - Action handler
  - Actions
  - Standard UI pattern
---
#  How to Create a Pick and Execute Process
  
##  Overview

**Pick and Execute (P&E)** is a case of  **Process Definition** with **Standard UI pattern**.

This section will add a new Pick and Execute process and associate it with the **Sales Order window**.

The implementation requires development experience. 

The following concept pages provide background information on action handlers and javascript

> development:

Reviewer: ```suggestion
development:

```

>   * [Action Handler](how-to-create-client-event-handler-actions.md)

Reviewer: ```suggestion

```

  * Fill the required fields 
  * Select **Pick and Execute** as window type 

![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_a_Pick_and_Execute_Process-1.png)

  * Following the same concepts, a **Table** is required as data-source. 

    In case it is needed to mix information from several tables, there are different options:

      * A database view used to create a table in [Application Dictionary](how-to-create-a-table.md).

>       

Reviewer: ```suggestion
```

  * Set the Handler (Java class implementing the process) 
  * Save 

![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_a_Pick_and_Execute_Process-6.png)

#####  Parameters

Window Reference is an implementation of a **rich parameter**.
   
  * Move to Parameters tab 
  * Create a new record 
  * Fill the required fields. The name of the parameter will be the name shown in the title of the running process. 
  * Select Window Reference 
  * Pick the Window you defined previously 
  * Save 

![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_a_Pick_and_Execute_Process-7.png)

####  Adding a button to Sales Order

#####  Create a Column

A new column is required to be associated to a button.

!!!info

>       For more information, read [How to add a field to a Window Tab](../how-to-guides/how-to-add-a-field-to-a-window-tab).

Reviewer: ```suggestion
      For more information, visit [How to add a field to a Window Tab](how-to-add-a-field-to-a-window-tab).
```
